src/Interpreters/DatabaseAndTableWithAlias.cpp

```diff
@@ -37,7 +38,7 @@ DatabaseAndTableWithAlias::DatabaseAndTableWithAlias(const ASTIdentifier & ident
     else if (identifier.name_parts.size() == 1)
         table = identifier.name_parts[0];
     else
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Logical error: invalid identifier");
+        throw Exception(ErrorCodes::INVALID_IDENTIFIER, "Invalid identifier");
```

Comment:
What exactly is invalid? How can a user fix their query to make it valid? Since it's not a logical error, and a user may see it, we should make the exception message clear

Reply:
@yakov-olkhovskiy

Reply:
this is pretty much clear from example I had provided:
```SELECT t.t.t.* FROM system.tables```
I don't see it requires any clarification - such error just some sort of typo

Reply:
Imagine a very very long query with subqueries, joins, complex functions, tons of different identifiers, etc...
It will be nearly impossible to understand something from the error message like this

Reply:
You are right - it worth to add identifier itself to the error message - I'll make a PR...